apl-core/src/main/java/com/apollocurrency/aplwallet/apl/exchange/utils/TradingViewUtils.java

```diff
@@ -0,0 +1,185 @@
+                
+                if ( currentPairRate.compareTo(hi) == 1 ) {
+                    hi = currentPairRate;
+                }
+
+                if ( currentPairRate.compareTo(low) == -1 ) {
+                    low = currentPairRate;
+                } 
+
+                BigDecimal amount = EthUtil.fromAtm( BigDecimal.valueOf(entryOfPeriod.getOrderAmount()) );
+                BigDecimal vx = amount.multiply(entryOfPeriod.getPairRate());
+                
+                log.trace("amount: {}, rate: {}", amount, entryOfPeriod.getPairRate() );
+                
+                volumefrom = volumefrom.add(amount);
+                volumeto = volumeto.add(vx);
+            }            
+            result.low = low; 
+            result.high = hi;
+            result.open = open; 
+            result.close = close;             
+            result.volumefrom = volumefrom;
+            result.volumeto = volumeto;            
+        } else {
+            result.low = BigDecimal.ZERO;
+            result.high = BigDecimal.ZERO;
+            result.open = BigDecimal.ZERO;
+            result.close = BigDecimal.ZERO;             
+            result.volumefrom = BigDecimal.ZERO;
+            result.volumeto = BigDecimal.ZERO;     
+        }
+        return result;         
+    }        
+
+
+        
+    static public TradingDataOutput getDataForIntervalFromOffers( String fsym, String tsym, Integer toTs, Integer limit, Integer interval, DexService service, TimeService timeService) {
+
+        int initialTime = toTs - (interval*limit);
+        int startGraph = initialTime;
+
+            
+        long startTS = (long)initialTime * 1000L;
+        long endTS = (long)toTs * 1000L; 
+            
+        Integer toTS = Convert2.toEpochTime(startTS);
+                                   
+        byte currencyType = 0;
+        
+        if ( tsym.equals("ETH") || fsym.equals("ETH") ) {
+            currencyType = 1;
+        } 
+                
+        if ( tsym.equals("PAX") || fsym.equals("PAX") ) {
+            currencyType = 0;
+        } 
+                                        
+        log.debug("start: {}, finish: {}, currencyType: {}, requestedType: {}", startTS, endTS, currencyType ); 
+                        
+        Integer startTSEpoch = Convert2.toEpochTime(startTS);
+        Integer endTSEpoch = Convert2.toEpochTime(endTS);
+            
+        log.debug("Epoch, start: {}, finish: {}", startTSEpoch, endTSEpoch ); 
+            
+        DexOrderDBRequestForTrading dexOrderDBRequestForTrading = new DexOrderDBRequestForTrading(startTSEpoch, endTSEpoch, (byte)1, currencyType, 0 , Integer.MAX_VALUE);
+            
+        List<DexOrder> dexOrdersForInterval = service.getOrdersForTrading(dexOrderDBRequestForTrading); 
+        
+        log.debug("found {} orders", dexOrdersForInterval.size() );
+        for (DexOrder cr : dexOrdersForInterval) {
```

Comment:
There is a bulk,time-consuming operation. Please, wrap it using log.isTraceEnabled.